bazel/linux/defs.bzl

```diff
@@ -259,6 +259,16 @@ def _kernel_modules(ctx):
         modules = " ".join(modules),
     )
 
+    compilation_mode = ctx.var["COMPILATION_MODE"]
```

Comment:
nothing to do: the hipster/modern way to do this would be to use //config like you were before, but I'm told the only way to do so from a rule... is to wrap it in a macro, something like:
```
_real = rule(
  attrs = {
    "opt_mode": attr.string(...)
  }..
)

def wrapper(...):
   _real(..., opt_mode = select({"//:is_opt": "opt", "//:is_fastbuild": "fastbuild", "//:is_dbg": "dbg"})
```

where //:is_X is a special target that  is used in select that allows to check if a config parameter is set.

(I'm fine with your current implementation, this is too much work)